### PR TITLE
refactor(utils/yaml): centralize YAML parsing to allow central templating handling

### PR DESCRIPTION
**Pull request by secustor**
## Changes
Moves YAML parsing and template string removal to a util function. 

## Context
This is in preparation to support templates in more YAML based package files such as HelmV3 or Kubernetes

https://github.com/renovatebot/renovate/discussions/24477

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/yaml.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+export function load(content: string, options?: YamlOptions): unknown {
+  const massagedContent = massageContent(content, options);
+  return single(massagedContent, options);
+}
+
+export function dump(obj: any, opts?: DumpOptions | undefined): string {
+  return upstreamDump(obj, opts);
+}
+
+export function massageContent(content: string, options?: YamlOptions): string {
```
Comment (viceice) [inline lib/util/yaml.ts:33]: do we really need to export this? it's already called by load.

File: lib/util/yaml.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  content: string,
+  iterator?: null | undefined,
+  options?: YamlOptions,
+): unknown[] {
+  const massagedContent = massageContent(content, options);
+
+  return multiple(massagedContent, iterator, options);
+}
+
+export function load(content: string, options?: YamlOptions): unknown {
```
Comment (viceice) [inline lib/util/yaml.ts:24]: can we rename it as load is really common. I think we should match the JSON parseJson

**Review by viceice — COMMENTED** [review]
we can do the rename in a follow up pr

File: lib/util/yaml.ts (hunk 1 above)
Comment (secustor) [reply lib/util/yaml.ts]: I have planned to write separate tests for it, but haven't done that in the end.

**Review by viceice — APPROVED** [review]

**Pull request merged**